app/services/save_practice_service.rb

```diff
@@ -393,4 +400,47 @@ def process_risk_mitigations_params
       end
     end
   end
+
+  def update_practice_editors
+    editors = @practice_params[:practice_editors_attributes]
+    includes_destroy = editors.keys.include?('_destroy')
+    if includes_destroy
+      editor_count = @practice.practice_editors.count
+      # delete practice editor if there is at least one left after deletion. If not, raise an error.
+      if editor_count > 1
+        PracticeEditor.find_by(id: editors[:id]).destroy
+      else
+        raise StandardError.new "error. At least one editor is required"
+      end
+    else
+      email_param = editors.values.first.values.first
+      user = User.find_by(email: email_param)
+      practice_editor = PracticeEditor.find_by(practice: @practice, user: user)
+
+      # if the user tries to save with a blank email field, raise an error
+      raise StandardError.new "error. Email field cannot be blank" if email_param.blank?
+
+      if user.present? && practice_editor.nil?
+        # create a new practice editor for the existing user
+        PracticeEditor.create_and_invite(@practice, user, email_param)
+      # make sure a duplicate practice editor is not created
+      elsif user.present? && practice_editor.present?
+        raise StandardError.new "error. A user with the email \"#{user.email}\" is already an editor for this practice"
+      else
+        # create a new user if they do not exist
+        user = User.new(email: email_param)
```

Comment:
not something to change right now... but just want to discuss for potential future cleanup ideas: thoughts on creating something like a UserService to create new users 🤔 💭

Reply:
I think that would be a good idea. I can look into adding one for this PR.

Reply:
All right, I added a `user` module and a method to make these calls reusable.